src/core/library.c

```diff
@@ -91,12 +96,38 @@ static void _bindConstraints(sqlite3_stmt* statement, const struct mLibraryEntry
 		sqlite3_bind_int(statement, useIndex, 1);
 		sqlite3_bind_int(statement, index, constraints->platform);
 	}
+
+	if (constraints->platformModels != M_LIBRARY_MODEL_UNKNOWN) {
+		index = sqlite3_bind_parameter_index(statement, ":models");
```

Comment:
the `useIndex` equivalent as well as actually updating the CONSTRAINTS definition seem to be missing